app/src/main/java/unipd/se18/ocrcamera/AllergenListAdapter.java

```diff
@@ -0,0 +1,103 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.CompoundButton;
+import android.widget.TextView;
+import android.widget.ToggleButton;
+
+import java.util.ArrayList;
+
+/*
+ * This class is used to initiate the listViews in MainAllergensActivity
+ * Also defined here are the instructions performed when a user selects a new allergen from the list
+ * @author Pietro Balzan
+ */
+public class AllergenListAdapter extends ArrayAdapter<Allergen> {
+
+    private static final String TAG = "AllergenListAdapter";
+    private Context mContext;
+    private int mResource;
+    private AllergensManager mAllergensManager;
+    private ArrayList<Allergen> selectedAllergens;
+
+    /**
+     * Default constructor for an AllergensListAdapter
+     * @param context of the calling activity
+     * @param resource int of the resource layout file of the view to adapt
+     * @param objects an arraylist of objects to be adapted
+     */
+    AllergenListAdapter(Context context, int resource, ArrayList<Allergen> objects) {
+        //adapt the Allergens List "objects"
+        super(context, resource, objects);
+        mContext = context;
+        //in our practical case we will be using allergen_single.xml
+        mResource = resource;
+        //set an AllergenManager used to modify users' allergens list
+        mAllergensManager= new AllergensManager(mContext);
+        selectedAllergens = mAllergensManager.getSelectedAllergensList();
+    }
+
+    /**
+     * this method deals with the adaptation of the single allergen view
+     * to the ListView of AllergensSearchFragment
+     * @param position of the Allergen
+     * @param convertView the view to be adapted
+     * @param parent viewGroup
+     * @return convertView the adapted View
+     */
+    @NonNull
+    @Override
+    public View getView(final int position, View convertView, ViewGroup parent) {
+        try {
+            //Allergen information
+            final Allergen mAllergen= getItem(position);
```

Comment:
I think what happens here is that since we are overriding getView the program automatically uses the list of Allergens ("objects") that was passedto the "  super(context, resource, objects);" at line 37. So i never wrote a "getItem(int i)" method and the one you see is just a standard method that works on ArrayLists of any object.